packages/perspective-workspace/src/ts/workspace/workspace.ts

```diff
@@ -1054,4 +1188,29 @@ export class PerspectiveWorkspace extends SplitPanel {
             );
         }
     }
+
+    /// swaps all tables in the workspace with the ones passed in.
+    /// Any viewers that are using tables not in the new tables will be ejected (`viewer>eject()`).
+    async replaceTables(
+        newTables: Record<string, psp.Table | Promise<psp.Table>>
+    ) {
+        const viewers = this.node.querySelectorAll(
+            "perspective-viewer"
+        ) as NodeListOf<HTMLPerspectiveViewerElement>;
+        // eject all viewers that have a table not in the new set of tables.
+        for (const viewer of viewers) {
+            if (!this._last_updated_state) break;
+            const t = this._last_updated_state.viewers[viewer.slot]?.table;
+            if (!newTables[t]) {
+                await viewer.eject();
+            } else {
+                await viewer.load(newTables[t]);
```

Comment:
Please batch these and await them all at the end using `Promise.all()`, so the user does not have to watch a slide show of these elements updating.